src/OpenTelemetry.Collector.Dependencies/AzureSdkCollector.cs

```diff
@@ -0,0 +1,136 @@
+﻿// <copyright file="AzureSdkCollector.cs" company="OpenTelemetry Authors">
+// Copyright 2018, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+namespace OpenTelemetry.Collector.Dependencies
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Diagnostics;
+    using OpenTelemetry.Context;
+    using OpenTelemetry.Trace;
+
+    public class AzureSdkCollector : IDisposable, IObserver<DiagnosticListener>, IObserver<KeyValuePair<string, object>>
+    {
+        private readonly ITracer tracer;
```

Comment:
there is already `DependenciesCollector` that takes care about sampler, tracer and subscription. Did you find any reason to not reuse it  (and also inherit this one from `ListenerHandler`)

Reply:
ListenerHandler seems to be slightly specific to HttpClientHandler listener

Reply:
Done